dev/doc/build-system.dune.md

```diff
@@ -10,7 +10,8 @@ Coq can now be built using [Dune](https://github.com/ocaml/dune).
 
 ## Quick Start
 
-You need Dune >= 1.2.1 ; just type `dune build` to build the base Coq
+Dune >= 1.5.0 is recommended, see `dune-project` for the minimum
+required version; type `dune build` to build the base Coq
```

Comment:
It is a bit confusing to start like this given that calling `dune build` on a clean tree will error (one must first call `makefile -f Makefile.dune voboot`). On the other hand `dune build _build/install/default/bin/coqtop` will work for instance.

Reply:
I thought of that; my rationale was that in the future it will just work.

By the way, it doesn't error in my system unless I have coq installed by OPAM. I dunno why. But `dune build` should definitively work even today and build coq without plugins and libraries indeed.

Reply:
```
File "dev/dune", line 7, characters 25-41:
7 |  (libraries coq.toplevel coq.plugins.ltac))
                             ^^^^^^^^^^^^^^^^
Error: Library "coq.plugins.ltac" not found.
Hint: try: dune external-lib-deps --missing @@default
```

Reply:
```
$ dune external-lib-deps --missing @@default
Done: 0/0 (jobs: 0)Error: The following libraries are missing in the default context:
- coq.plugins.ltac
Hint: try: opam install coq
```

Reply:
`dune build _build/install/default/bin/coqtop` works though

Reply:
And same for `coqide`, `coqidetop.opt`.

Reply:
Ahh, good point; this got broken recently; will push a fix by declaring the dev libraries optional. Native Coq support will make this problem go fully away anyways.

Reply:
>I thought of that; my rationale was that in the future it will just work.

We should document what works in the present, the future can document itself later.

Reply:
Fixed in #9035